File: 프로그래머스/0/181893. 배열 조각하기/배열 조각하기.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public int[] solution(int[] arr, int[] query) 
    {
        List<int> arrList = new List<int>(arr);
        int count = 0;
        
        for (int i=0; i<query.Length; i++) {
            
            if(i%2 == 0) {
                while(arrList.Count > query[i]+1) {
                    arrList.RemoveAt(query[i]+1);
                }
            }
            
            else {
                while(count < query[i]) {
                    arrList.RemoveAt(0);
                    count++;
                }
                count = 0;
            }
        }
        
        int[] answer = arrList.ToArray();
        return answer;
    }
}

File: 프로그래머스/0/181893. 배열 조각하기/배열 조각하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<int> query) 
{
    // begin() => 0번째 인덱스를 가르킴
    // end() => 마지막 인덱스 + 1를 가르킴 [,) 
    for(int i=0; i<query.size(); i++)
    {
        if(i%2 == 0)
        {
            arr.erase(arr.begin() + query[i] + 1, arr.end()); // query[i] + 1번째 인덱스부터 마지막 인덱스까지 삭제
        }
        else
        {
            arr.erase(arr.begin(), arr.begin() + query[i]); // 0번째 인덱스부터 query[i]번째 인덱스 전까지 삭제
        }
    }
    return arr;
}